examples/lua_REPL/main.c

```diff
@@ -0,0 +1,50 @@
+/*
```

Comment:
maybe this could be in the same `lua` example. What do you think?

Reply:
(basically the LUA Repl implementation is running LUA. So, should be enough to show how it works)

Reply:
Yes, the thing is the REPL requires more resources, so it runs on less boards (part of that is because I'm enabling the lua test module).
My idea was to have the lua example showcase the basic API, and another example with a more complex app.

Reply:
check the OpenThread test at #7149. It's possible to switch between Full Thread Device and Minimum Thread Device with a Makefile flag.

Reply:
Are you suggesting I test the type of device and run the REPL or the Hello World accordingly?

Reply:
yes, something like that. Like:

```
/* call hello world */
#ifdef MODULE_LUA_REPL
/* call Lua repl module */
#endif
```

Reply:
Then I will have two programs in one, which is what happens each time one has an `#ifdef`

Reply:
Yes. I tend to prefer one example showing key functionalities than several. It's only one opinion though, so maybe would be nice to hear other people here too.

Reply:
I think it should be two separate examples if it is already done that way.  More obvious when looking at examples.  +1 to original implementation.

Reply:
ok, let's leave it like that then. We can discuss it later in further PRs and don't block this one

Reply:
(or what @jcarrano thinks the best)

Reply:
+1 for separate examples for clarity reasons. Could rename `example/lua` to `example/lua_hello-world` or something. In general I think it's more important that the examples are all concise, useful and non-overlapping than that they are simply small in number

Reply:
I like @danpetry approach

Reply:
This is corrected